### DISCUSSION TITLE
index.js source code is displaying instead of deployed application on link?

### DISCUSSION THREAD

**Discussion by KowyXYZ** [vercel/community#1555] [Help] [answered]
### Summary

Hi, I tried to deploy the app on Vercel without any custom addons except adding custom path to source, and after deployment its only showing source code from my index.js and nothing else...  btw My app is running perfectly fine when I run it from local host using react.

Thanks <3

Website link: https://e-commerce-repo-ten.vercel.app

**Comment by amyegan** [comment]
Hi @KowyXYZ! Are you able to share your repo or a minimal reproduction?

**Reply by KowyXYZ** [reply]
Of course i will make it public

**Comment by KowyXYZ** [comment, marked as answer]
So i fixed it, problem was in nodemodules for some reason, so to fix this create **.gitignore** file with path to nodemodules and thats it. 
Be sure that your root path is correct

**Comment by jonathaneemmett** [comment]
@amyegan I have a .gitignore and still its only showing the code.  I even copied and pasted code from the vercel website to see if it was just something I was missing.  Here is the repo.  https://github.com/jonathaneemmett/vercel-api-demo

**Reply by jonathaneemmett** [reply]
This doesn't look like rocket science, so apparently I am not a rocket scientist. ..lol

**Reply by amyegan** [reply]
Hi @jonathaneemmett! I'm not able to run your code to see the problem. But it looks like you have an Express app, so I recommend reading through our guide on [using Express.js with Vercel](https://vercel.com/guides/using-express-with-vercel) and making the suggested adjustments to get it working in a Serverless Function.

**Reply by jonathaneemmett** [reply]
Hi @amyegan, actually it came down to an issue with express which that article doesn't cover.   

instead of:

`
app.use(express.json());
app.use(express.urlencoded({ extended: true }));
`

I had to install body-parser and use:

`
app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));
`

**Comment by Danishsoma** [comment]
https://my-doctor-react-ixe8.vercel.app/ is showing the index.js source code instead of the application. Can anyone help
@amyegan @KowyXYZ

**Reply by amyegan** [reply]
Hi @Danishsoma. Please [start a new discussion](https://github.com/vercel/community/discussions/new) with more information about your project. If you can include a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) or your public repo, that is often useful for collaborative debugging.
